src/common/hooks/useEndorsementChain/useEndorsementChain.ts

```diff
@@ -51,3 +48,30 @@ export const useEndorsementChain = (
 
   return { endorsementChain, pending, error };
 };
+
+type ErrorWithMessage = {
+  message: string;
+};
+
+export const isErrorWithMessage = (error: unknown): error is ErrorWithMessage => {
+  return (
+    typeof error === "object" &&
+    error !== null &&
+    "message" in error &&
+    typeof (error as Record<string, unknown>).message === "string"
+  );
+};
+
+export const toErrorWithMessage = (maybeError: unknown): ErrorWithMessage => {
+  if (isErrorWithMessage(maybeError)) return maybeError;
+
+  try {
+    return new Error(JSON.stringify(maybeError));
+  } catch {
+    return new Error(String(maybeError));
+  }
+};
+
+export const getErrorMessage = (error: unknown): string => {
+  return toErrorWithMessage(error).message;
+};
```

Comment:
Should this be in a common place so other places that handles error can use it too.